opensearch-project/OpenSearch-Dashboards
# [Workspace]Disable confirm button during collaborators operation

### Description

This PR addresses an issue where the confirm modal was closed after initiating collaborator updates or deletions, allowing users to perform other operations while the API call was in progress. Previously, if changes were made during the API call, the last operation could be missed or overwritten.

To resolve this issue, the following changes have been made:

1. The confirm button is now disabled during collaborator updates and deletions to prevent additional operations from being triggered while the API call is in progress.

~~2. The confirm modal remains open during collaborator updates and deletions, preventing users from performing other operations until the current operation is completed.~~ Done by #8594 

By keeping the confirm modal open and disabling the confirm button, the user interface now accurately reflects the ongoing operation and prevents potential data loss or inconsistencies caused by concurrent operations.


## Screenshot


![image](https://github.com/user-attachments/assets/0578e79c-2ca7-4087-9aab-216e3a93ac34)
![image](https://github.com/user-attachments/assets/833dff48-596b-45f3-80f0-78f0acac2947)


## Testing the changes



- Clone branch code and run `yarn osd bootstrap --single-version loose`
- Add below configs in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
opensearchDashboards.dashboardAdmin.users: ['admin']
```
- Run `yarn start --no-base-path`
- Login with admin user
- Visit an existing workspace setting page
- Add some users or groups by "Add collaborators" button
- Check the left checkbox in some table rows
- Click "Actions" button at the top right of table
- Click "Change access level"
- Click "Admin"
- Click "Confirm" button and it should be disabled during updating
- Click "Delete x collaborator" button at the top left of table
- Click "Confirm" button in popped confirm modal
- The "Confirm" button should be disabled during deleting

## Changelog

- feat: [Workspace]Keep confirm modal and disable confirm button during collaborators operation

### Check List
